sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -120,53 +118,185 @@ public async Task<ModuleIdentity> CreateModuleIdentityAsync(DeviceIdentity devic
+                moduleIdentity.ManagedBy = Environment.UserName;
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.CreateOrUpdateIdentityAsync(moduleIdentity, IfMatchPrecondition.UnconditionalIfMatch);
+
+                ModuleIdentity updatedModule = response.Value;
+                
+                SampleLogger.PrintSuccess($"Successfully updated module identity: DeviceId: '{deviceId}', ModuleId: '{updatedModule.ModuleId}', ManagedBy: '{updatedModule.ManagedBy}', ETag: '{updatedModule.Etag}'");
+                
+                return updatedModule;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to update a module identity due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task UpdateModuleTwinAsync()
+        /// <summary>
+        /// Get module twin.
+        /// </summary>
+        /// <param name="deviceId">Device identity unique identifier the module belongs to.</param>
+        /// <param name="moduleId">Module identity unique identifier to be updated.</param>
+        public async Task<TwinData> GetModuleTwinAsync(string deviceId, string moduleId)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("GET A MODULE TWIN");
+
+            try
+            {
+                Console.WriteLine($"Getting module twin with Id: '{moduleId}'");
+
+                Response<TwinData> response = await IoTHubServiceClient.Modules.GetTwinAsync(deviceId, moduleId);
+
+                SampleLogger.PrintSuccess($"\t- Module Twin: DeviceId: '{deviceId}', ModuleId: '{moduleId}', Status: '{response.Value.Status}', ETag: '{response.Value.Etag}'");
+
+                return response.Value;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to get a module twin due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task InvokeMethodAsync()
+        /// <summary>
+        /// Update a module twin desired properties
+        /// </summary>
+        /// <param name="deviceId">Device identity unique identifier the module belongs to.</param>
+        /// <param name="moduleId">Module identity unique identifier to be updated.</param>
+        public async Task<TwinData> UpdateModuleTwinAsync(string deviceId, string moduleId)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("UPDATE A MODULE TWIN");
+
+            string userPropName = "user";
+
+            try
+            {
+                // Get the device module
+                Response<TwinData> getResponse = await IoTHubServiceClient.Modules.GetTwinAsync(deviceId, moduleId);
+                TwinData moduleTwin = getResponse.Value;
+
+                Console.WriteLine($"Updating module twin: DeviceId: '{moduleId}', ModuleId: '{moduleTwin.ModuleId}', ETag: '{moduleTwin.Etag}'");
```

Comment:
Did you mean to write {moduleId} --> {deviceId} and for {moduleTwin.ModuleId} being {moduleId}?

Reply:
they are the same thing, but I will change it to be more clear.